### PR TITLE
Add Translate API samples. Fixes #176.

### LINKED ISSUE #176: Translate API samples
- [x] - translate.js
  - [x] - Samples
    - [x] - detect
    - [x] - getLanguages
    - [x] - translate
  - [x] - CLI
  - [x] - Unit tests
  - [x] - System tests

### PR DESCRIPTION
**Pull request by jmdobry**

### CODE REVIEW

**Comment by jmdobry** [comment]
@ace-n When you have a moment.

File: translate/README.md [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+
+```
+Commands:
+  detect <text>     Detect the language of the provided text
+  list              List available translation languages.
+  translate <text>  Translate the provided text to the target language.
+
+Options:
+  --help     Show help                                                 [boolean]
+  --key, -k                                                  [string] [required]
```
Comment (ace-n) [inline translate/README.md]: Explain what this is/does? (I'm guessing it's an API key?)

File: translate/README.md [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+
+Examples:
+  node translate detect "Hello world!"      Detect the language of "Hello
+                                            world!".
+  node translate list                       List available translation
+                                            languages.
+  node translate translate --to ru "Good    Translate "Good morning!" to
```
Comment (ace-n) [inline translate/README.md]: Include `key` in this example, if it's necessary?

File: translate/test/translate.test.js [hunk 3]
```diff
@@ -0,0 +1,188 @@
...
+
+describe('translate:translate', function () {
+  describe('detectLanguage', function () {
+    it('should detect language', function () {
+      var sample = getSample();
+      var callback = sinon.stub();
+
+      sample.program.detectLanguage(text, key, callback);
+
+      assert(sample.mocks.translate.detect.calledOnce, 'create called once');
```
Comment (ace-n) [inline translate/test/translate.test.js]: s/create/detect/ ?

File: translate/test/translate.test.js [hunk 4]
```diff
@@ -0,0 +1,188 @@
...
+        text: text,
+        to: 'ru',
+        key: key
+      };
+
+      sample.program.translateText(options, callback);
+
+      assert(sample.mocks.translate.translate.calledOnce, 'method called once');
+      assert.equal(sample.mocks.translate.translate.firstCall.args.length, 3, 'method received 3 arguments');
+      assert.deepEqual(sample.mocks.translate.translate.firstCall.args[0], text, 'method received correct first argument');
```
Comment (ace-n) [inline translate/test/translate.test.js]: Should this be a `deepEqual`?

File: translate/translate.js [hunk 5]
```diff
@@ -0,0 +1,169 @@
...
+// [START all]
+// [START setup]
+// By default, the client will authenticate using the service account file
+// specified by the GOOGLE_APPLICATION_CREDENTIALS environment variable and use
+// the project specified by the GCLOUD_PROJECT environment variable. See
+// https://googlecloudplatform.github.io/gcloud-node/#/docs/google-cloud/latest/guides/authentication
+var Translate = require('@google-cloud/translate');
+
+// Helper library for language codes
+var ISO6391 = require('iso-639-1');
```
Comment (ace-n) [inline translate/translate.js]: Is it worth giving our _var_ for this a better name?

File: translate/test/translate.test.js (hunk 4 above)
Comment (jmdobry) [inline translate/test/translate.test.js]: Fixed.

File: translate/test/translate.test.js (hunk 3 above)
Comment (jmdobry) [inline translate/test/translate.test.js]: Fixed.

File: translate/translate.js [hunk 6]
```diff
@@ -0,0 +1,169 @@
...
+  .option('key', {
+    alias: 'k',
+    global: true,
+    demand: true,
+    requiresArg: true,
+    type: 'string'
+  })
+  .example('node $0 detect "Hello world!"', 'Detect the language of "Hello world!".')
+  .example('node $0 list', 'List available translation languages.')
+  .example('node $0 translate --to ru "Good morning!"', 'Translate "Good morning!" to Russian, auto-detecting English.')
```
Comment (ace-n) [inline translate/translate.js]: Include API key in this one?

**Comment by ace-n** [comment]
LGTM aside from a few minor nits.

File: translate/README.md (hunk 2 above)
Comment (jmdobry) [inline translate/README.md]: Fixed.

File: translate/README.md (hunk 1 above)
Comment (jmdobry) [inline translate/README.md]: Fixed.

File: translate/translate.js (hunk 6 above)
Comment (jmdobry) [inline translate/translate.js]: Fixed.

File: translate/translate.js (hunk 5 above)
Comment (jmdobry) [inline translate/translate.js]: Like what? That's what the README for `iso-639-1` uses.

Comment (ace-n) [inline translate/translate.js]: I was thinking something like `languagecodes` - but either way, it's not that big a deal.

**Comment by ace-n** [comment]
LGTM!

**Pull request merged**